compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/symtab_transformer/gen_c_transformer/name_transformer.rs

```diff
@@ -0,0 +1,245 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use super::super::super::{
+    DatatypeComponent, Expr, Location, Parameter, Stmt, Symbol, SymbolTable, Type,
+};
+use super::super::Transformer;
+use rustc_data_structures::fx::{FxHashMap, FxHashSet};
+
+/// Struct for replacing names with valid C identifiers for --gen-c-runnable.
+pub struct NameTransformer {
+    new_symbol_table: SymbolTable,
+    mapped_names: FxHashMap<String, String>,
+    used_names: FxHashSet<String>,
+}
+
+impl NameTransformer {
+    /// Transform all identifiers in the symbol table to be valid C identifiers.
+    pub fn transform(original_symbol_table: &SymbolTable) -> SymbolTable {
+        let new_symbol_table = SymbolTable::new(original_symbol_table.machine_model().clone());
+        NameTransformer {
+            new_symbol_table,
+            mapped_names: FxHashMap::default(),
+            used_names: FxHashSet::default(),
+        }
+        .transform_symbol_table(original_symbol_table)
+    }
+
+    /// Converts an arbitrary identifier into a valid C identifier.
+    fn normalize_identifier(&mut self, orig_name: &str) -> String {
+        assert!(!orig_name.is_empty(), "Received empty identifier.");
+
+        // If name already encountered, return same result
```

Comment:
Comment explaining that this isn't just an optimization, its required for correctness

Reply:
Done.